src/olm/account/mod.rs

```diff
@@ -682,16 +792,163 @@ mod libolm {
     }
 }
 
+mod dehydrated_device {
+    use matrix_pickle::{Decode, DecodeError, Encode, EncodeError};
+    use zeroize::{Zeroize, ZeroizeOnDrop};
+
+    use super::{
+        fallback_keys::{FallbackKey, FallbackKeys},
+        one_time_keys::OneTimeKeys,
+        Account,
+    };
+    use crate::{
+        types::{Curve25519Keypair, Curve25519SecretKey},
+        DehydratedDeviceError, Ed25519Keypair, KeyId,
+    };
+
+    #[derive(Encode, Decode, Zeroize, ZeroizeOnDrop)]
+    pub(crate) struct OneTimeKey {
+        pub(crate) private_key: Box<[u8; 32]>,
```

Comment:
I know this is a bit useless since you are boxing the value, but we should use the `#[secret]` annotation anyways: https://github.com/matrix-org/matrix-pickle/?tab=readme-ov-file#encoding-and-decoding-secrets.

Reply:
I added those myself.